### PR TITLE
Gutenberg: refactor copied edit-post folder

### PR DESCRIPTION
**Pull request by vindl**
#### Changes proposed in this Pull Request

* Remove copied `edit-posts` components that are now exported as part of the package.
* Removes `assets` folder and replaces it with built styles import.

#### Testing instructions

1. Navigate to http://calypso.localhost:3000/gutenberg/
2. Verify that the editor shell still works as expected and that there are no new console errors.
3. Verify that all usages of components that `edit-post` package exports are now replaced.

Fixes https://github.com/Automattic/wp-calypso/issues/26642 (to the extent that `edit-post` package now allows)
Helps with https://github.com/Automattic/wp-calypso/pull/28069

### CODE REVIEW

File: assets/stylesheets/sections/gutenberg-editor.scss [hunk 1]
```diff
@@ -24,31 +24,8 @@
...
-@import 'gutenberg/editor/edit-post/components/sidebar/sidebar-header/style';
-@import 'gutenberg/editor/edit-post/components/sidebar/post-trash/style';
-@import 'gutenberg/editor/edit-post/components/sidebar/block-sidebar/style';
-@import 'gutenberg/editor/edit-post/components/visual-editor/style';
-@import 'gutenberg/editor/edit-post/components/header/style';
-@import 'gutenberg/editor/edit-post/components/header/header-toolbar/style';
-@import 'gutenberg/editor/edit-post/components/header/more-menu/style';
-@import 'gutenberg/editor/edit-post/components/header/pinned-plugins/style';
+// Edit-post component styles
+@import '../../../node_modules/@wordpress/edit-post/build-style/style';
```
Comment (vindl) [inline assets/stylesheets/sections/gutenberg-editor.scss]: Props @mmtr for the suggestion!

**Comment by vindl** [comment]
@Copons I'm seeing some visual issues with `Close` button and Ground controls. Did some of their CSS end up in `edit-post/assets` folder?

**Comment by vindl** [comment]
@gziolo if there are more components that could be easily exposed from `edit-post` (that wouldn't require significant refactor), it would help us with further reducing the size of the copied code that we need to maintain further:

https://github.com/WordPress/gutenberg/blob/3c81345cd8a8f9bb7879bea2b304464c10ddd896/packages/edit-post/src/index.js#L88-L93

File: client/gutenberg/editor/style.scss [hunk 2]
```diff
@@ -108,4 +108,35 @@ $gutenberg-theme-toggle: #11a0d2;
 			z-index: 100000; // Above WP toolbar.
 		}
 	}
+
+	.edit-post-header-toolbar__back.is-borderless {
```
Comment (vindl) [inline client/gutenberg/editor/style.scss]: Restoring this from `gutenberg/editor/edit-post/components/header/header-toolbar/style`. Originally introduced in https://github.com/Automattic/wp-calypso/pull/26809/.

**Comment by mmtr** [comment]
I noticed a change in the font of the post content. 

Previously it was using the "Noto Serif":
screen shot 2018-10-25 at 18 05 17

But now it's using the default system font:
screen shot 2018-10-25 at 18 05 25

I guess [this `font-family` definition](https://github.com/Automattic/wp-calypso/blob/5c5c85f1b6cc70a6bf853ae4c7decc54a42dd75e/client/gutenberg/editor/edit-post/components/visual-editor/style.scss#L8) is missing somewhere.

**Comment by gwwar** [comment]
Great line count here ✨ Let's try to squash the major noticeable issues, land this, and then follow up with other bugs in follow up issues/PRs.

One thing I found a bit annoying was that I can't scroll the content body with a mouse-wheel here vs on wpcalypso. eg I'm only able to scroll content when my mouse is over the sidebar/top bar vs anywhere else on the editor. Are other folks able to reproduce this?

screen shot 2018-10-25 at 11 03 52 am

More minor, but we do see a visible scrollbar in FF on both master/this branch

screen shot 2018-10-25 at 11 11 51 am

**Comment by vindl** [comment]
> One thing I found a bit annoying was that I can't scroll the content body with a mouse-wheel here vs on wpcalypso. eg I'm only able to scroll content when my mouse is over the sidebar/top bar vs anywhere else on the editor. Are other folks able to reproduce this?

I can't reproduce this. I suspected it was something related to very long Archives block, but I can't repro in that case either.

**Comment by gwwar** [comment]
> I can't reproduce this. I suspected it was something related to very long Archives block, but I can't repro in that case either.

Hmm okay I won't let that be a blocker then 👍

**Comment by vindl** [comment]
Thanks @gwwar, I'll land this as is and handle the remaining issues in follow up PRs.

**Comment by gziolo** [comment]
> @gziolo if there are more components that could be easily exposed from edit-post (that wouldn't require significant refactor), it would help us with further reducing the size of the copied code that we need to maintain further:

We would rather move some of the block to `@wordpress/editor` package. Can you identify those which would fit there? I'm sure that we will have to do it in phase 2 when working on customization, cc @youknowriad.

**Pull request merged**